networkx/algorithms/centrality/betweenness.py

```diff
@@ -357,25 +362,61 @@ def _accumulate_edges(betweenness, S, P, sigma, s):
 
 
 def _rescale(betweenness, n, normalized, directed=False, k=None, endpoints=False):
-    if normalized:
+    # The non-normalized BC values are computed the same way for directed and
+    # undirected graphs: shortest paths are computed and counted for each
+    # *ordered* (s, t) pair. Undirected graphs should only count valid
+    # *unordered* node pairs {s, t}; that is, (s, t) and (t, s) should
+    # be counted only once.
+    if endpoints and n < 2 or not endpoints and n <= 2:
+        # no normalization b=0 for all nodes
+        scale = None
+    elif normalized:
+        # The numerator is computed the same for directed and undirected
+        # graphs, so the normalization factor is the same too.
         if endpoints:
-            if n < 2:
-                scale = None  # no normalization
-            else:
-                # Scale factor should include endpoint nodes
+            # Scale factor should include endpoint nodes.
+            # Scale by all (s, t) combos where s != t.
+            if k is None:
                 scale = 1 / (n * (n - 1))
-        elif n <= 2:
-            scale = None  # no normalization b=0 for all nodes
+            else:
+                scale = 1 / (k * (n - 1))
         else:
-            scale = 1 / ((n - 1) * (n - 2))
-    else:  # rescale by 2 for undirected graphs
-        if not directed:
-            scale = 0.5
+            # Exclude endpoints and count the number of (s, t) node pairs
+            # that could have a path pass through v where v is not s or t.
+            if k is None:
+                scale = 1 / ((n - 1) * (n - 2))
+            else:
+                scale = 1 / (k * (n - 2))
```

Comment:
> Fix edge case when k=n and endpoints is False

Here is where this fix comes into play. If `k = n`, these scales would be different for each of these branches. If `k = n`, the `k is None` branch should be used, because the `else` branch violates the assumption that `v` is not `s` or `t`.